Sources/_StringProcessing/Compiler.swift

```diff
@@ -1,7 +1,7 @@
-import MatchingEngine
+import _MatchingEngine
 
 public struct RegexProgram {
```

Comment:
Internal too, as the compiler and executor aren't user-facing.

Reply:
But `Executor` is, and `Compiler` is used by the regex consumer conformance

Reply:
Alternatively, we pull that stuff in to this module too, or else have simple public functions to do compilation etc.

Reply:
Oh... Then I guess we can leave them public for now until we find a way to better organize the algorithms.

Reply:
I went ahead and dropped the whole public VirtualMachine thing and TortoiseVM used as a reference. We'll just have to actually test the matching engine going forwards :-)